sqlglot/dataframe/sql/column.py

```diff
@@ -0,0 +1,330 @@
+from __future__ import annotations
+
+import datetime
+import typing as t
+from collections.abc import Iterable
+
+import sqlglot
+from sqlglot import expressions as exp
+from sqlglot.dataframe.sql.types import DataType
+from sqlglot.helper import flatten
+
+if t.TYPE_CHECKING:
+    from sqlglot.dataframe.sql._typing import (
+        ColumnOrName,
+        ColumnOrPrimitive,
+        DateTimeLiteral,
+        DecimalLiteral,
+        Literals,
+        Primitives,
+    )
+    from sqlglot.dataframe.sql.window import WindowSpec
+
+
+class Column:
+    def __init__(self, expression: t.Union[t.Any, exp.Expression]):
+        if isinstance(expression, str):
+            expression = sqlglot.parse_one(expression, read="spark")
+        elif isinstance(expression, Column):
+            expression = expression.expression
+        self.expression = expression
```

Comment:
Do we need to ensure the validity of the passed expression in any way? I assume not every expression can be `Column`.

Reply:
Many can. I think I'm actually good with this approach for now and see if we get any feedback about issues. My guess is that this won't be an issue.